src/frontend/components/LiveVideoPanel/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect } from 'react';
+import { Tabs, Box, Grommet } from 'grommet';
+
+import { Chat } from 'components/Chat';
+import {
+  LivePanelItem,
+  useLivePanelState,
+} from 'data/stores/useLivePanelState';
+import { Video } from 'types/tracks';
+import { ShouldNotHappen } from 'utils/errors/exception';
+import { breakpointSize, theme } from 'utils/theme/theme';
+
+import { LiveVideoTabPanel } from './LiveVideoTabPanel';
+
+interface LiveVideoPanelProps {
+  video: Video;
+}
+
+export const LiveVideoPanel = ({ video }: LiveVideoPanelProps) => {
+  const { currentItem, availableItems, setPanelVisibility } = useLivePanelState(
+    (state) => ({
+      currentItem: state.currentItem,
+      availableItems: state.availableItems,
+      setPanelVisibility: state.setPanelVisibility,
+    }),
+  );
+
+  //  close panel if there is nothing to display
+  useEffect(() => {
+    if (!currentItem) {
+      setPanelVisibility(false);
+    }
+  }, [currentItem]);
+
+  if (!currentItem) {
+    return <React.Fragment />;
+  }
+
+  let header;
+  if (availableItems.length > 1) {
+    header = (
+      <Tabs
+        activeIndex={availableItems.indexOf(currentItem)}
+        onActive={(index) => setPanelVisibility(true, availableItems[index])}
+      >
+        {availableItems.map((item) => (
+          <LiveVideoTabPanel
+            item={item}
+            selected={item === currentItem}
+            key={`tab_${item}`}
+          />
+        ))}
+      </Tabs>
+    );
+  }
+
+  let content;
+  switch (currentItem) {
+    case LivePanelItem.CHAT:
+      content = <Chat video={video} />;
+      break;
+    case LivePanelItem.APPLICATION:
+      //  TODO : implement this item
+      content = <p>application content</p>;
+      break;
+    case LivePanelItem.JOIN_DISCUSSION:
+      //  TODO : implement this item
+      content = <p>join discussion content</p>;
+      break;
+    default:
+      throw new ShouldNotHappen(currentItem);
```

Comment:
```suggestion
      throw new ShouldNotHappen(currentItem);
```
Can you test this?

Reply:
I'm not sure to understand what to test here.

This will never be throw because typescript will raise an error (that's the purpose of typing it with never)

Reply:
I was wondering, If it can't be tested and can never happen, what's the point then of the code?

Reply:
adding new values in the enum not handle in a case will raise a typescript error and then prevent the code from transpiling